pkg/power/components/source/rapl_msr_util.go

```diff
@@ -85,21 +83,20 @@ func mapPackageAndCore() error {
 		if err != nil {
 			return err
 		}
-		if packageMap[id] == -1 {
-			packageMap[id] = i
-		}
-		if maxPackage < id {
-			maxPackage = id
-		}
+
```

Comment:
Using the package id instead of the array position looks like a good approach.

I don't know the details, but the original code might have some reason to do not update the packageMap[i] when it is not null, i.e., `!=-1`.

So, I would keep this logic.

Reply:
Well, I don't know too much detail either, but from code perspective, the original `packageMap` is an array and adopt index as content of `physical_package_id`, but in my env, the `physical_package_id` is not as its expectation(equal to cpu core index, 0 - 15), so I adopt `map`, I guess I cannot keep the original logic, I mean '!=-1'.
based on the original logic: 
```
if packageMap[id] == -1 {
    packageMap[id] = i
}
```
seems, it try to avoid reset, does that means the `physical_package_id` may has duplicated value?